### DISCUSSION TITLE
asset/source not working with dynamic string

### DISCUSSION THREAD

**Discussion by D-Thrane** [webpack/webpack#16637] [Q&A]
# Bug report

**What is the current behavior?**

I have a create-react-app test site. I did the following in webpack.config.js

```
test: /\.svg$/,
type: 'asset/source',
```
When I do this I get the svg xml source as intended.

```
const logo = require(`./logo.svg`);
console.log("logo: ", logo);
```

But when I do this I get "/static/media/logo.6ce24c58023cc2f8fd88.svg".

```
const logoName = "logo";
const logo = require(`./${logoName}.svg`);
console.log("logo: ", logo);
```

**If the current behavior is a bug, please provide the steps to reproduce.**

Require svg from a dynamic string like so:
```
const logoName = "logo";
const logo = require(`./${logoName}.svg`);
console.log("logo: ", logo);
```

**What is the expected behavior?**

Expect to get the svg xml source.

**Other relevant information:**
webpack version:  5.64.4
Node.js version: 9.2.0
Operating System: macOS 13.0.1 (22A400)
Additional tools:

**Comment by alexander-akait** [comment]
Hello, it is expected, because webpack using static analization to get import we can't analize your variable and undestand your import, please read https://webpack.js.org/api/module-methods/#dynamic-expressions-in-import and feel free to ask questions

**Reply by D-Thrane** [reply]
How would a dynamic require then look?

**Reply by alexander-akait** [reply]
Do you realy need dynamic request? You can use `/* webpackIgnore: true */` and use dynamic `import(...)`, you can put files in directory and use `import(`./directory/${name}.svg`)`, so webpack can handle it

**Reply by D-Thrane** [reply]
Why I am getting this when i try. `Failed to load module script: Expected a JavaScript module script but the server responded with a MIME type of "image/svg+xml". Strict MIME type checking is enforced for module scripts per HTML spec.`

**Reply by alexander-akait** [reply]
It means dev server returns 404 HTML page (look at HTML code), maybe you rename it while bundling or it was just nit copied, don't forget when you bundle application you will have all modules in one file (except async part/configuration), so if you use used `import("../../../image.svg")`, it will not work, because:
- you need copy them to `public`/`dist` (based on your application structure) directory
- use `import("/images/image.svg")` (just example)

Other wise you will have in `dist/main.js` path to `../../../image.svg` and it is invalid path

**Reply by D-Thrane** [reply]
But if I look in network, it has loaded the svg with a 200.

**Reply by alexander-akait** [reply]
Are you sure? Can you provide small reproducible example, I think that you missing soemthing small and we will fix it easy

**Reply by D-Thrane** [reply]
Yes. looks like network get's the image ok.

![image](https://user-images.githubusercontent.com/24752553/225857566-6e0e1893-2f18-4149-82ca-aec4a18c0f56.png)

I did create a new app with create-react-app. And then did the eject to get the webpack.config. Dont know if that is the problem?

I have changed the svg test in webpack.config to this:
```
{
  test: /\.svg$/,
  type: 'asset/source',
}
```

My component code looks like this:
```
import(/* webpackIgnore: true */`/svg-embed/${iconName}.svg`).then((content) => {
  console.log("content: ", content);
  return(setSvgContent(content.default));
});
```

The image is in public/svg-embed/image.svg

When i build the image ends up in build/svg-embed/image.svg

**Reply by alexander-akait** [reply]
Hm, don't know, if you provide me github repo with the problem I will help, but CRA has a lot of weird value for options under the hood, so it is hard to say
